haskell/repl.bzl

```diff
@@ -74,6 +76,7 @@ HaskellReplCollectInfo = provider(
     fields = {
         "load_infos": "Dictionary from labels to HaskellReplLoadInfo.",
         "dep_infos": "Dictionary from labels to HaskellReplDepInfo.",
+        "java_deps": "depset of Files to jars needed for building.",
```

Comment:
Could this be carried in `HaskellReplLoadInfo` instead?

The idea here is that `HaskellReplCollectInfo` collects "load from source" and "load from binary" information separately for each target associated to the target's label. This information is then combined into `HaskellReplInfo` later on based on the `from_source` and `from_binary` patterns. Adding `java_deps` directly here changes that design. Is there a good reason to do so?

---

If I understand correctly then JARs are only compile time dependencies for targets loaded from source to interpret inline-java TH splices. With targets loaded from binary they are only runtime dependencies and so should only be made available if `collect_data = True` to be consistent. Does that make sense?

Reply:
> If I understand correctly then JARs are only compile time dependencies for targets loaded from source to interpret inline-java TH splices. 

JARs are both compile and runtime dependencies in this case.

> With targets loaded from binary they are only runtime dependencies and so should only be made available if collect_data = True to be consistent.

This makes sense.

Reply:
I still don't understand the propose rearrangement.

If I move `java_deps` to `HaskellReplLoadInfo`, how do I keep track of the jars that need to be added to the classpath when the target is loaded from binary?

Reply:
hm, strictly speaking, the CLASSPATH doesn't need to be set when running. But it would be a convenient way to signal to the program the list of jar paths to feed to the JVM when initializing it.

[This issue](https://github.com/tweag/inline-java/issues/179) is related. Maybe we should handle this concern separately.

Reply:
I moved `java_deps` to `HaskellReplLoadInfo`.

Reply:
> If I move `java_deps` to `HaskellReplLoadInfo`, how do I keep track of the jars that need to be added to the classpath when the target is loaded from binary?

Right, so they would also be needed in `HaskellReplDepInfo` for the runtime deps case.

Reply:
> hm, strictly speaking, the CLASSPATH doesn't need to be set when running. But it would be a convenient way to signal to the program the list of jar paths to feed to the JVM when initializing it.
> 
> [This issue](https://github.com/tweag/inline-java/issues/179) is related. Maybe we should handle this concern separately.

Oh, I remember. Yes, happy to treat this separately.